db-engine/src/main/java/com/splicemachine/db/iapi/sql/dictionary/fk/Graph.java

```diff
@@ -29,12 +36,66 @@ public Graph(Set<String> vertices) {
             vertexIndex.put(vertex, i++);
         }
         parents = new int[vertices.size()][];
+        surrogateCounter = 0;
+        this.newConstraintName = newConstraintName;
+        this.surrogates = new HashMap<>();
+    }
+
+    void addEdge(String from, String to, EdgeNode.Type type) throws StandardException {
```

Comment:
Maybe add some comments describing the graph structure. Having an example with three nodes would be very helpful in understanding how it works. Currently, interface doesn't match call site (from <-> to reversed) and it's a bit confusing.

Reply:
done.

Reply:
totally agree, I fixed it, and added more documentation.